packages/blast-utilities/contract-utils.js

```diff
@@ -4,97 +4,80 @@ const {
 } = require('cudosjs')
 const path = require('path')
 const fs = require('fs')
-
-const { getGasPrice } = require('./config-utils.js')
-const { getAccountAddress } = require('./account-utils')
-const { getClient } = require('./client.js')
+const { getGasPrice } = require('./config-utils')
 const BlastError = require('./blast-error')
-
-const Contract = class {
-  constructor(contractname, initMsg, label) {
-    this.contractname = contractname
-    this.initMsg = initMsg
-    this.label = label || contractname
-    this.client = {}
+const { getProjectRootPath } = require('./package-info')
+
+module.exports.CudosContract = class CudosContract {
+  #contractName
+  #owner
+  #contractAddress
+  #wasmPath
+  #gasPrice
+  
+  constructor(contractName, owner, deployedContractAddress = null) {
+    this.#contractName = contractName
+    this.#owner = owner
+    this.#contractAddress = deployedContractAddress
+    this.#wasmPath = path.join(getProjectRootPath(), `artifacts/${contractName}.wasm`)
+    this.#gasPrice = GasPrice.fromString(getGasPrice())
+
+    if (!this.#isDeployed() && !fs.existsSync(this.#wasmPath)) {
+      throw new BlastError(`Contract with name ${contractName} was not found, did you compile it?`)
+    }
   }
 
-  async init() {
-    this.client = getClient()
-
-    if (!this.deployed) {
-      this.wasmPath = path.join(process.cwd(), `artifacts/${this.contractname}.wasm`)
-      if (!fs.existsSync(this.wasmPath)) {
-        throw new BlastError(`Contract with name ${this.contractname} was not found, did you compile it?`)
-      }
+  async deploy(initMsg, owner = this.#owner, label = this.#contractName) {
+    if (this.#isDeployed()) {
+      throw new BlastError(`Contract is already deployed!`)
     }
+    
+    this.#owner = owner
+    const uploadReceipt = await this.#uploadContract()
+    console.log(uploadReceipt)
+    const ic = await this.#initContract(uploadReceipt.codeId, initMsg, label)
```

Comment:
can we use more meaningful const here like contractInstance or something like that